### PR TITLE
Remove special handling of single 'Any' parameter and other hacks

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

File: lib/Sema/CSRanking.cpp [hunk 1]
```diff
@@ -726,6 +726,19 @@ Comparison TypeChecker::compareDeclarations(DeclContext *dc,
   return decl1Better? Comparison::Better : Comparison::Worse;
 }
 
+static Type getUnlabeledType(Type type, ASTContext &ctx) {
+  if (auto *tupleType = dyn_cast<TupleType>(type.getPointer())) {
```
Comment (rudkx) [inline lib/Sema/CSRanking.cpp]: Looking forward to this one day being a `getAs<TupleType>`...

File: lib/Sema/CSSimplify.cpp [hunk 2]
```diff
@@ -1276,6 +1276,19 @@ ConstraintSystem::matchExistentialTypes(Type type1, Type type2,
...
+    if (flags.contains(TMF_GenerateConstraints)) {
+      addUnsolvedConstraint(
+        Constraint::create(*this, kind, type1, type2,
+                           getConstraintLocator(locator)));
+      return getTypeMatchSuccess();
+    }
+
+    return getTypeMatchAmbiguous();
+  }
+
```
Comment (rudkx) [inline lib/Sema/CSSimplify.cpp]: Glad you caught this!

**Review by rudkx — COMMENTED** [review]
LGTM!

**Pull request closed without merging**